File: src/emojisToCharacterCodes.ts
export const emojisToCharacterCodes = (template: string) =>
  template
    .replace(/🟥/g, "{63}")
    .replace(/🟧/g, "{64}")
    .replace(/🟨/g, "{65}")
    .replace(/🟩/g, "{66}")
    .replace(/🟦/g, "{67}")
    .replace(/🟪/g, "{68}")
    .replace(/⬜/g, "{69}")
    .replace(/⬛/g, "{70}")

    // Map heart variants to char 62, which Flagship boards render as a heart.
    // ❤️ (with U+FE0F variation selector) must be replaced before bare ❤.
    .replace(/❤️/g, "{62}")
    .replace(/❤/g, "{62}")
    .replace(/🧡/g, "{62}")
    .replace(/💛/g, "{62}")
    .replace(/💚/g, "{62}")
    .replace(/💙/g, "{62}")
    .replace(/💜/g, "{62}")
    .replace(/🖤/g, "{62}")
    .replace(/🤍/g, "{62}")
    .replace(/🤎/g, "{62}")

    // Handle special case for ß to SS
    .replace(/ß/g, "SS");

File: python/pyvbml/emojis_to_character_codes.py
"""Emojis to character codes.

Port of Vestaboard/vbml/src/emojisToCharacterCodes.ts
"""

from __future__ import annotations


def emojis_to_character_codes(template: str) -> str:
    """Emojis to character codes."""
    return (
        template.replace("🟥", "{63}")
        .replace("🟧", "{64}")
        .replace("🟨", "{65}")
        .replace("🟩", "{66}")
        .replace("🟦", "{67}")
        .replace("🟪", "{68}")
        .replace("⬜", "{69}")
        .replace("⬛", "{70}")
        # Heart variants → char 62 (rendered as a heart on Flagship boards).
        # ❤️ (with U+FE0F variation selector) must be replaced before bare ❤.
        .replace("❤️", "{62}")
        .replace("❤", "{62}")
        .replace("🧡", "{62}")
        .replace("💛", "{62}")
        .replace("💚", "{62}")
        .replace("💙", "{62}")
        .replace("💜", "{62}")
        .replace("🖤", "{62}")
        .replace("🤍", "{62}")
        .replace("🤎", "{62}")
        .replace("ß", "SS")  # special case for ß → SS
    )
